control-plane/typeSpecConverter.cpp

```diff
@@ -111,6 +115,48 @@ bool TypeSpecConverter::preorder(const IR::Type_Name* type) {
     return false;
 }
 
+// This function is invoked if psa.p4 has a Newtype. The function
+// is not invoked for user-defined NewType.
+bool TypeSpecConverter::preorder(const IR::Type_Newtype* type) {
+    if (p4RtTypeInfo) {
+        bool orig_type = true;
+        const IR::StringLiteral* uri;
+        const IR::Constant* sdnB;
+        auto ann = type->getAnnotation("p4runtime_translation");
+        if (ann != nullptr) {
+            orig_type = false;
+            uri = ann->expr[0]->to<IR::StringLiteral>();
+            BUG_CHECK(uri, "P4runtime annotation does not have uri: %1%",
+                      type->getP4Type());
+            sdnB = ann->expr[1]->to<IR::Constant>();
+            BUG_CHECK(sdnB, "P4runtime annotation does not have sdn: %1%",
```

Comment:
This should be an error, not a BUG; the user can supply an incorrect expression, and this should not crash the compiler.